conjure_oxide/tests/integration/basic/weighted-sum/02-needs-normalising/input-expected-rule-trace-human.txt

```diff
@@ -0,0 +1,54 @@
+-(1), 
+   ~~> apply_eval_constant ([("Constant", 9001)]) 
+-1 
+
+--
+
+-(5), 
+   ~~> apply_eval_constant ([("Constant", 9001)]) 
+-5 
+
+--
+
+Sum([Sum([Sum([Sum([Product([5, x]), Product([Product([Product([y, 3]), 1]), 3])]), -(Product([3, x]))]), Product([-1, y])]), Product([y, -5])]), 
+   ~~> normalise_associative_commutative ([("Base", 8900)]) 
+Sum([Product([5, x]), Product([Product([Product([y, 3]), 1]), 3]), -(Product([3, x])), Product([-1, y]), Product([y, -5])]) 
+
+--
+
+Product([Product([Product([y, 3]), 1]), 3]), 
+   ~~> normalise_associative_commutative ([("Base", 8900)]) 
+Product([y, 3, 1, 3]) 
+
+--
+
+Product([y, 3, 1, 3]), 
+   ~~> partial_evaluator ([("Base", 9000)]) 
+Product([y]) 
```

Comment:
what happened here? should it not have been `Product([y, 9])`?

Reply:
Fixed. I set my counter to 0 initially instead of 1.